pkg/utils/utils.go

```diff
@@ -375,6 +376,17 @@ func configSriovDevice(iface *sriovnetworkv1.Interface, ifaceStatus *sriovnetwor
 	return nil
 }
 
+func generateRandomMacAddressVfs() ([6]byte, error) {
+	buf := make([]byte, 6)
+	_, err := rand.Read(buf)
+	if err != nil {
+		return nil, err
+	}
+	// Set the local bit
+	buf[0] = (buf[0] & 0xfe) | 2
```

Comment:
From the Kernel:
/**
 * eth_random_addr - Generate software assigned random Ethernet address
 * @addr: Pointer to a six-byte array containing the Ethernet address
 *
 * Generate a random Ethernet address (MAC) that is not multicast
 * and has the local assigned bit set.
 */
static inline void eth_random_addr(u8 *addr)
{
	get_random_bytes(addr, ETH_ALEN);
	addr[0] &= 0xfe;	/* clear multicast bit */
	addr[0] |= 0x02;	/* set local assignment bit (IEEE802) */
}

Could you expand the commentary to explain your bit operations?

Reply:
Yes, I agree that it is the same. But your commentary "// Set the local bit" could be improved to explain the bit operations in more detail.

Reply:
I can 

> > iI a way (besides checking the mac) that the card is **purposefully** not setting the MAC? We've had other cases where the mac was not set unintentionally. With your changes, the mac will be set either way complicating the debug process.
> 
> This is a very good point. As per the further discussions, it seems that this is a Pensando specific issue. I think we have the following options:
> 
> 1. Create the random MAC addresses in the Kernel driver for Pensando cards. Similar to the IAVF drivers.
> 2. Create a PensandoPlugin for this specific issue to isolate the changes to Pensando cards only.
> 3. Keep this PR as is. However the con is what Balazs was quoted above.
> 
> Did we go over these options and decide that option 3 is the best approach?

Discussed with @bn222 offline. We will look into this as a seperate issue.